bee-network/src/tests/common/keys_and_ids.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::PeerId;
+
+pub fn gen_constant_peer_id() -> PeerId {
+    "12D3KooWJWEKvSFbben74C7H4YtKjhPMTDxd7gP7zxWSUEeF27st".parse().unwrap()
+}
+
+#[cfg(feature = "full")]
+mod __full {
+
+    use super::*;
+    use libp2p::identity::{
+        ed25519::{Keypair, SecretKey},
+        PublicKey,
+    };
+    use std::iter::repeat;
+
+    pub fn gen_random_peer_id() -> PeerId {
+        PeerId::from_public_key(libp2p_core::PublicKey::Ed25519(Keypair::generate().public()))
+    }
+
+    pub fn gen_deterministic_peer_id(gen: impl ToString) -> PeerId {
+        let keys = gen_deterministic_keys(gen);
+        PeerId::from_public_key(PublicKey::Ed25519(keys.public()))
+    }
+
+    pub fn gen_deterministic_keys(gen: impl ToString) -> Keypair {
+        let hex_chars = ('0'..'9').chain('a'..'f').collect::<Vec<_>>();
+        let gen = gen.to_string();
+        let gen2 = gen.to_string();
+        let chars = gen2.chars();
+
+        for c in gen.chars() {
+            if !hex_chars.contains(&c) {
+                panic!("invalid generator");
+            }
+        }
+
+        let div = 64 / gen.len();
+        let rem = 64 % gen.len();
+
+        let identity_sk = repeat(gen)
+            .take(div)
+            .chain(chars.into_iter().map(Into::into).take(rem))
+            .collect::<String>();
+
+        let mut hex_sk = hex::decode(identity_sk).unwrap();
```

Comment:
Why not use this instead of the validation check that's being performed above? <inserts URL to 'parse, don't check' blog post>

Reply:
not sure...I am not parsing here. I am generating the secret key deterministically myself from a string that is required to only contain valid hex chars. So for example you give it "123abc", and it will create "123abc123abc123abc...123a" (without bc at the end!) which is a valid Ed25519 secret key.

Reply:
ah... now I understand what you mean. Sure can change that.